src/Components/WebAssembly/WebAssembly/src/Hosting/MutableConfiguration.cs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Hosting
+{
+    /// <summary>
+    /// MutableConfiguration is a class that implements the interface of an IConfiguration,
+    /// IConfigurationRoot, and IConfigurationBuilder. It can be used to simulatneously build
+    /// and read from a configuration object.
+    /// </summary>
+    public class MutableConfiguration : IConfiguration, IConfigurationRoot, IConfigurationBuilder
```

Comment:
We should pick a different name given that this is part of the public API surface. Suggestion `WebAssemblyHostConfigurationBuilder`

Reply:
Renamed this to `WebAssemblyHostConfiguration`.